platforms/defaults_windows.go

```diff
@@ -39,25 +39,28 @@ func DefaultSpec() specs.Platform {
 	}
 }
 
-type matchComparer struct {
-	defaults        Matcher
+type windowsmatcher struct {
+	specs.Platform
 	osVersionPrefix string
+	defaultMatcher  Matcher
 }
 
 // Match matches platform with the same windows major, minor
 // and build version.
-func (m matchComparer) Match(p imagespec.Platform) bool {
-	if m.defaults.Match(p) {
-		// TODO(windows): Figure out whether OSVersion is deprecated.
-		return strings.HasPrefix(p.OSVersion, m.osVersionPrefix)
+func (m windowsmatcher) Match(p imagespec.Platform) bool {
+	match := m.defaultMatcher.Match(p)
+
+	if match && m.OS == "windows"  {
+		return strings.HasPrefix(p.OSVersion, m.osVersionPrefix) && m.defaultMatcher.Match(p)
```

Comment:
Windows Server 2022 has made this more complex, as its kernel (`10.0.20348.x`) is now a stable ABI, and so Windows hosts with greater kernel builds (only currently Windows 11, `10.0.22000.y), can run container images marked `10.0.20348.z` in process isolation mode, and hence should pass this test.

More details at https://github.com/microsoft/Windows-Containers/issues/117#issuecomment-912645956, https://techcommunity.microsoft.com/t5/containers/windows-server-2022-and-beyond-for-containers/ba-p/2712487, and https://github.com/Microsoft/hcsshim/blob/master/test/cri-containerd/main_test.go#L162-L212.

And also https://github.com/microsoft/Windows-Containers/issues/170#issuecomment-1024520322, where I recently discovered the issue being addressed here and in #6497, and forgot to raise a bug.

(On that note, if we want to punt the down-level support into a new containerd issue, that's fine too.)

Reply:
my vote is to handle this in separate issue.

Reply:
Done as #6508.